### PR TITLE
add support for port=<port> in a hostfile for plm/rsh

### PR DESCRIPTION
**Pull request by ggouaillardet**
Fixes open-mpi/ompi#2224

### CODE REVIEW

File: orte/mca/plm/rsh/plm_rsh_module.c [hunk 1]
```diff
@@ -1222,6 +1223,16 @@ static void launch_daemons(int fd, short args, void *cbdata)
         } else {
             argv[node_name_index1] = strdup(node->name);
         }
+        portptr = &port;
+        if (orte_get_attribute(&node->attributes, ORTE_NODE_PORT, (void**)&portptr, OPAL_INT)) {
+            char *portname;
+            fprintf(stderr, "port = %d\n", port);
```
Comment (rhc54) [inline orte/mca/plm/rsh/plm_rsh_module.c]: can you please remove this debug?

File: orte/util/hostfile/hostfile.c [hunk 2]
```diff
@@ -156,6 +156,7 @@ static int hostfile_parse_line(int token, opal_list_t* updates,
         argv = opal_argv_split (value, '@');
 
         cnt = opal_argv_count (argv);
+        fprintf(stderr, "cnt=%d\n", cnt);
```
Comment (rhc54) [inline orte/util/hostfile/hostfile.c]: can you please remove this debug?

File: orte/util/hostfile/hostfile.c [hunk 3]
```diff
@@ -335,12 +336,23 @@ static int hostfile_parse_line(int token, opal_list_t* updates,
...
             }
             break;
 
+        case ORTE_HOSTFILE_PORT:
+            rc = hostfile_parse_int();
+            if (rc < 0) {
+                fprintf(stderr, "hostfile: NEGATIVE port=%d\n", rc);
+                return ORTE_ERROR;
+            }
+            fprintf(stderr, "hostfile: port=%d\n", rc);
```
Comment (rhc54) [inline orte/util/hostfile/hostfile.c]: please remove debug?

File: orte/util/hostfile/hostfile_lex.l [hunk 4]
```diff
@@ -121,12 +121,17 @@ slot               { orte_util_hostfile_value.sval = yytext;
                      return ORTE_HOSTFILE_SLOT; }
 
 username           { orte_util_hostfile_value.sval = yytext;
+                     fprintf(stderr, "lex: username=%s\n", yytext);
```
Comment (rhc54) [inline orte/util/hostfile/hostfile_lex.l]: please remove debug?

File: orte/util/hostfile/hostfile_lex.l [hunk 5]
```diff
@@ -121,12 +121,17 @@ slot               { orte_util_hostfile_value.sval = yytext;
...
                      return ORTE_HOSTFILE_USERNAME; }
 "user-name"        { orte_util_hostfile_value.sval = yytext;
                      return ORTE_HOSTFILE_USERNAME; }
 "user_name"        { orte_util_hostfile_value.sval = yytext;
                      return ORTE_HOSTFILE_USERNAME; }
 
+port               { orte_util_hostfile_value.sval = yytext;
+                     fprintf(stderr, "lex: port=%s\n", yytext);
```
Comment (rhc54) [inline orte/util/hostfile/hostfile_lex.l]: please remove debug?

**Comment by ggouaillardet** [comment]
@rhc54 i removed all the debug, and added an error message if a negative port is specified

**Pull request merged**